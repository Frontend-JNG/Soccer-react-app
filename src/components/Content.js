import React, { useState } from "react";
import "../App.css";
import Leagues from "./Leagues";
import Standings from "./Standings";
import Button from "@mui/material/Button";

import Grid from "@mui/material/Grid";

const Content = () => {
  const [active, setActive] = useState(true);

  return (
    <Grid
      container
      spacing={2}
      display="flex"
     
      style={{ justifyContent: "center", marginTop: "5px" }}
    >
      <Grid item xs={2}>
        <Button
          fullWidth
          color="primary"
          variant="contained"
          onClick={() => setActive(true)}
        >
          <h2 style={{ color: active ? "secondary" : null }}>Leagues</h2>
        </Button>
      </Grid>
      <Grid item xs={2}>
        <Button
          fullWidth
          color="primary"
          variant="contained"
          onClick={() => setActive(false)}
        >
          <h2 style={{ color: !active ? "secondary" : null }}>Standings</h2>
        </Button>
      </Grid>
      {active ? <Leagues /> : <Standings />}
    </Grid>
  );
};

export default Content;
