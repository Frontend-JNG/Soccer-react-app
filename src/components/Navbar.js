import React from "react";
import "../App";
import { Link } from "react-router-dom";
import ContactForm from "../contactForm";

const Navbar = () => {
  return (
    <nav className="navbar">
      <div className="navtext">
        <h1>Football App âš½</h1>
      </div>
      <div className="nav-links">
        <li>
          <Link className="links" to="/">
            Home
          </Link>
        </li>
        <li>
          <Link className="links" to="/Standings">
            Standings
          </Link>
        </li>
        <li>
          <Link className="links" to="/contactForm">
            Subscription Option
          </Link>
        </li>

        {/* <Link to="/">Home</Link>
          <Link to="/Standings"> Standings</Link>
          <Link to="/contactForm"> Contact</Link> */}
      </div>
    </nav>
  );
};

export default Navbar;
