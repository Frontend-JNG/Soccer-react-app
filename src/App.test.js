import { cleanup, render, screen } from '@testing-library/react';
import renderer from "react-test-renderer";
import Standings from './components/Standings';
import ContactForm from './contactForm';
import "@testing-library/jest-dom/extend-expect";
import Content from './components/Content';
import Navbar from './components/Navbar';

afterEach(()=>{
  cleanup();
});

it("standings and league's name", () => {
  const standings = {id: 364, location: "Liverpool", name: "Liverpool"}
  render(<Standings standings={standings} />);
  })
it("standings and league's name in the list", () => {
  const contactForm = { type: {}, name: "fName", id: "fName" };
  render(<ContactForm contactForm={contactForm} />);
  const contactForms = screen.getByTestId({id: "fName", name: "fName" });
  expect(contactForms).toBeInTheDocument()
  // expect(todoElement).toBe(jhon)
});
it("test", ()=>{
  expect(true).toBe(true)
})  

test('matches snapshot', () =>{
  const standings = { id: 364, location: "Liverpool", name: "Liverpool" };
  const tree = renderer.create(<Standings standings={standings} />).toJSON();
  expect(tree).toMatchSnapshot();
})


it('Should render same test passed into titile props', () => {
  render(<Navbar h1Element="Football App ⚽️ " />);
  const h1Element = screen.getByText(/football app ⚽️/i);
  expect(h1Element).toBeInTheDocument();
});

it('Should render components passed by role', () => {
  render(<Content title="Leagues"/>);
  const headingElement = screen.getByRole("heading", {name: "Leagues"});
  expect(headingElement).toBeInTheDocument();
});

it('Should render components passed by spesifc role', () => {
  render(<Content title="Standings"/>);
  const headingElement = screen.getByRole("heading", {name: "Standings"});
  expect(headingElement).toBeInTheDocument();
});




