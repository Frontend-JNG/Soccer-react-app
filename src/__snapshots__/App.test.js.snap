// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="standings-container"
>
  <div
    className="select-fields"
  >
    <select
      className="soccer"
      defaultValue="eng.1"
      id="select-league"
      name="select-league"
      onChange={[Function]}
    >
      <option
        value="arg.1"
      >
        Argentine Liga Profesional de Fútbol
      </option>
      <option
        value="aus.1"
      >
        Australian A-League
      </option>
      <option
        value="bra.1"
      >
        Brazilian Serie A
      </option>
      <option
        value="chn.1"
      >
        Chinese Super League
      </option>
      <option
        value="ned.1"
      >
        Dutch Eredivisie
      </option>
      <option
        value="eng.1"
      >
        English Premier League
      </option>
      <option
        value="fra.1"
      >
        French Ligue 1
      </option>
      <option
        value="ger.1"
      >
        German Bundesliga
      </option>
      <option
        value="ita.1"
      >
        Italian Serie A
      </option>
      <option
        value="jpn.1"
      >
        Japanese J League
      </option>
      <option
        value="mys1"
      >
        Malaysian Super League
      </option>
      <option
        value="mex.1"
      >
        Mexican Liga BBVA MX
      </option>
      <option
        value="por.1"
      >
        Portuguese Liga
      </option>
      <option
        value="esp.1"
      >
        Spanish Primera División
      </option>
    </select>
    <select
      className="soccer"
      defaultValue="2021"
      id="select-year"
      name="select-year"
      onChange={[Function]}
    >
      <option
        value="2011"
      >
        2011
      </option>
      <option
        value="2012"
      >
        2012
      </option>
      <option
        value="2013"
      >
        2013
      </option>
      <option
        value="2014"
      >
        2014
      </option>
      <option
        value="2015"
      >
        2015
      </option>
      <option
        value="2016"
      >
        2016
      </option>
      <option
        value="2017"
      >
        2017
      </option>
      <option
        value="2018"
      >
        2018
      </option>
      <option
        value="2019"
      >
        2019
      </option>
      <option
        value="2020"
      >
        2020
      </option>
      <option
        value="2021"
      >
        2021
      </option>
    </select>
  </div>
  <div
    className="ranking-div"
  />
</div>
`;
